### PR TITLE
Orphan file signals

### PR DESCRIPTION
**Pull request by rebbrunner**
Adds two signals, one to delete orphaned files when updating an image, and one to delete orphaned files when deleting a profile.

### CODE REVIEW

File: website/models.py [hunk 1]
```diff
@@ -61,3 +61,31 @@ def createUserProfile(sender, instance, created, **kwargs):
...
+
+    try:
+        oldPhoto = Profile.objects.get(pk=instance.pk).photo
+    except Profile.DoesNotExist:
+        return False
+
+    newPhoto = instance.photo
+    if os.path.basename(oldPhoto.name) == "defaultuser.png":
+        return False
+    else:
```
Comment (psybers) [inline website/models.py]: else is not needed here because the true branch returns

File: website/models.py [hunk 2]
```diff
@@ -61,3 +61,31 @@ def createUserProfile(sender, instance, created, **kwargs):
...
 def saveUserProfile(sender, instance, **kwargs):
     instance.profile.save()
+
+
+@receiver(post_delete, sender=Profile)
+def deleteOnDelete(sender, instance, **kwargs):
+    if instance.photo:
+        if os.path.basename(instance.photo.name) != "defaultuser.png":
+            if os.path.isfile(instance.photo.path):
+                os.remove(instance.photo.path)
```
Comment (psybers) [inline website/models.py]: I'm thinking there is some possibilities of the remove causing exceptions. We should probably try/catch them and silently ignore them.

File: website/models.py [hunk 3]
```diff
@@ -61,3 +61,31 @@ def createUserProfile(sender, instance, created, **kwargs):
...
+        return False
+
+    newPhoto = instance.photo
+    if os.path.basename(oldPhoto.name) == "defaultuser.png":
+        return False
+    else:
+        if os.path.basename(oldPhoto.name) == os.path.basename(newPhoto.name):
+            return False
+
+        if os.path.isfile(oldPhoto.path):
```
Comment (psybers) [inline website/models.py]: Would it make sense to refactor the code and pull the common code out into a helper function? The common code being testing if the file is 'defaultuser.png' and testing if it exists, and removing it. And then the new try/catch around that.

**Review by psybers — CHANGES_REQUESTED** [review]

File: website/models.py (hunk 2 above)
Comment (rebbrunner) [reply website/models.py]: What exceptions are you thinking of?

**Comment by psybers** [comment]
change so that uploaded files go into separate directories for each user, based on their UID (e.g., /media/1/foo.jpg)

**Comment by psybers** [comment]
dont use the filename as uploaded, generate a hash for each file and use the hash as the filename

**Comment by rebbrunner** [comment]
This code probably needs to be merged after the image resizing PR because it refactors models.py

**Comment by psybers** [comment]
Why is it named `helperCheckDefault`?  That name doesn't indicate the seriousness of the function. Maybe something like `removeProfilePhoto` instead?

**Comment by psybers** [comment]
The code has errors now. You renamed the function, but not the calls to it.

**Review by psybers — APPROVED** [review]

**Pull request merged**